### ISSUE TITLE
Prepare Pis/Drivers For Display Screens

### ISSUE THREAD

**Issue by mildlyincompetent** [srobo/tasks#96]
- prep the image
- flash the SDs (for the borrowed ones, we need to review what's already on them and possibly re-flash them back that afterwards)
- generate list of MAC addresses and desired DNS entries

This follows from #76 and specifically https://github.com/srobo/tasks/issues/76#issuecomment-478331948 has the details of what we want where.

**Comment by kierdavis** [comment]
See also #96.

**Comment by RealOrangeOne** [comment]
Do we have a canonical list of where we're going to have displays? It would be nice if we can get at least 1 per team pit location

**Comment by Tyler-Ward** [comment]
see also #99 for list of pis needed

**Comment by PeterJCLaw** [comment]
The Pis we have are in storage (#183), along with their SD cards. There's not a lot we can really do here without the SDs.

**Comment by Tyler-Ward** [comment]
I believe they are in southampton.

**Comment by PeterJCLaw** [comment]
Huh, the inventory agrees. This can progress then. I suggest the steps (once we know what screens are needed) are:
1. update [srcomp-kiosk](https://github.com/PeterJCLaw/srcomp-kiosk/)'s config: 
  - manually edit https://github.com/PeterJCLaw/srcomp-kiosk/blob/master/pi_macs, then run `./macs-to-names.py` and commit the result
  - update the name of the compbox to have "2019" in it (rather than whatever the previous year is)
  - might need to also agree on what the fixed IP of the compbox is going to be too
2. build a fresh image around that (see https://github.com/PeterJCLaw/srcomp-kiosk/blob/master/INSTALL.md, though note that you almost certainly _don't_ need to use the "lite" image it suggests)
3. write that image onto each of the Pis
4. (optionally) boot each Pi and run the `update` script mentioned in the install file
5. label each Pi with its intended destination, for easy deployment

The boot-and-update step is optional as I always end up running the update script on all the Pis after deployment _anyway_.

**Comment by Tyler-Ward** [comment]
Does the script do anything other than set up a kiosk browser as there are pi images that are an off the shelf solution for this. e.g. https://github.com/guysoft/FullPageOS

**Comment by PeterJCLaw** [comment]
It does things like turning off the screensaver & power-saving stuff, plus setting up SSH access.
It also knows intrinsically what URL to point at.
The main unusual thing it does is configure the compbox as the NTP upstream for the box.
If someone would like to explore off-the-shelf solutions that might be a useful iteration, though the NTP configuration is a hard requirement.

Note that because the current solution is known and well understood any replacement would be super-low priority. Additionally: the screens are _not_ tested in Chromium, so their performance and behaviour would need testing if we switched (from Firefox).

**Comment by PeterJCLaw** [comment]
Just had a very quick look at https://github.com/guysoft/FullPageOS. It's interesting, however:
- it seems to assume only a single Pi per network (`fullpageos.local` being _the_ network name used)
- no apparent mechanism to change the passwords used (we'd want to do that for sanity reasons)
- supports Pi 2+; we'll be using Pi 1s.

**Comment by PeterJCLaw** [comment]
Blocked on the Pis making their way to me for configuration.

**Comment by PeterJCLaw** [comment]
Progress on this:
- `srcomp-kiosk` is up to date with the planned layout
- I have an image which is deployable on to the Pis, however I'm actually tempted to leave most of them as they are (they all have various previous deployments on and are fairly easily updatable by just grabbing the latest config, I _think_)
- I've checked that all the Pis boot and can show a useful page

I _haven't_ check that all the SDs are writeable. Given the experience with the compbox (#261), this is a possible failure mode, but I'm less worried because spare micro/SD cards are easy to get at short notice.

**Comment by PeterJCLaw** [comment]
Update: we managed to flash all the Pi SDs ths evening. They're now in a position to be deployed, but will need configuring after that.

Configuration is a case of `ssh`ing into each and running `./srcomp-kiosk/scripts/update`.

**Comment by PeterJCLaw** [comment]
On the topic of alternative kiosk apps: http://openkiosk.mozdevgroup.com/

**Comment by PeterJCLaw** [comment]
These are done; any further updates are part of the standard setup/fixing phase, rather than prep.
